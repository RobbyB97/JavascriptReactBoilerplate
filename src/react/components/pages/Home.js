/* Packages */
import React from 'react';


/* Component */
export class Home extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            loaded: false,
            loading: false
        };
    };


    componentWillMount() {
        this.load();
    };


    load = () => {
        const $pageLoader = document.querySelector('.page-loader');

        // Disable loader
        if ($pageLoader) {
            $pageLoader.setAttribute('data-loading', true);
            this.setState({
                loading: true
            });
        };

        // Fade in React app
        setTimeout(() => {
            $pageLoader.setAttribute('data-loaded', true);
            this.setState({
                loaded: true
            });
        }, 300);
    };


    render() {
        return (
            <div id="Home" 
                data-loading={this.state.loading}
                data-loaded={this.state.loaded}
            >
                <section className="Home__header">
                    <h2>React Redux Boilerplate v1</h2>
                    <h4>By Robby Bergers</h4>
                </section>

                <section className="Home__content">
                    <article>
                        <h3>
                            This Boilerplate comes bundled with
                        </h3>
                        <hr/>
                        <ul>
                            <li>React</li>
                            <li>React Router</li>
                            <li>Redux</li>
                            <li>SASS</li>
                            <li>Webpack</li>
                            <li>Node.js</li>
                            <li>Express</li>
                            <li>Source mapping</li>
                            <li>Jest</li>
                        </ul>
                    </article>
                    
                    <article>
                        <a 
                            className="Home__link"
                            href="https://github.com/RobbyB97/JavascriptReactBoilerplate"
                        >
                            View on GitHub
                        </a>
                    </article>

                    <article>
                        <p>
                            Creating software such as this boilerplate takes a lot of time, effort, attention to detail, and the occasional excellent anger management. If you found this boilerplate helpful, please consider <a href="https://paypal.me/robbybobby123">donating</a>.
                        </p>

                        <a
                            className="Home__link"
                            href="https://paypal.me/robbybobby123"
                        >
                            Donate
                        </a>
                    </article>
                </section>
            </div>
        );
    };
};


export default Home;