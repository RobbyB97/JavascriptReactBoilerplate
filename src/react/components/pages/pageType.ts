export interface PageType {
    page_ID__Set: (id: string) => void;
}