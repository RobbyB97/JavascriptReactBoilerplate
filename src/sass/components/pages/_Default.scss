#Default {
    @include page;
    padding-top: 0;
    font-family: $sansSerif;
}


.Default__header {
    background-color: $primaryColor;

}


.Default__headerContent {
    @include innerContent;
    padding-top: 70px;
    padding-bottom: 32px;

    h1 {
        font-size: 2.8em;
        color: $offWhite;
        margin: 0;
    }

    p {
        color: lighten($primaryColor, 22%);
    }

    a {
        text-decoration: underline;
        color: lighten($primaryColor, 22%);
    }


    @media #{$tablet_breakpoint} {
        width: 400px;

        h1 {
            font-size: 3em;
        }
    }

    @media #{$desktop_breakpoint} {
        width: 600px;

        h1 {
            font-size: 3.2em;
        }
    }
}


.Default__tagline {
    text-align: center;
    font-size: 1.2em;
    color: $primaryColor_dark;
}


.Default__links {

}


.Default__link {
    display: inline-block;
    
    &:not(:last-child) {
        margin-bottom: 1em;
    }
}


.Default__content {
    article {
        padding: 20px;
        text-align: center;

        p {
            font-size: 1.2em;
        }
        
        ul {
            margin: 0 auto;
            list-style-type: none;
            font-size: $mFont;
            padding-left: 0;
        }
    }

    @media #{$tablet_breakpoint} {
        article {
            width: 80%;
            margin: 0 auto;

            p {
                font-size: $mFont;
            }
        }
    }

    @media #{$desktop_breakpoint} {
        article {
            max-width: 1000px;
            margin: 0 auto;    
        }
    }
}


.Default__link {
    padding: 8px;
    text-decoration: none;
    background: $primaryColor;
    color: $offWhite;
    font-weight: bold;
    font-size: $mFont;
    margin: 0 4px;
    transition: 200ms;

    &:active, &:hover {
        background: $offWhite;
        color: $primaryColor;
    }
}


.Default__donate {
    opacity: 0.8;
    
    p {
        font-size: $sFont !important;
    }
}