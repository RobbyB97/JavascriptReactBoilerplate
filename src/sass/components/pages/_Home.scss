#Home {
    opacity: 0;
    background: $white;
    font-family: "Montserrat";
    
    &[data-loading=true] {
        &[data-loaded=true] {
            transition: opacity 300ms;
            opacity: 1;
            background: $white;
        }
    }
}


.Home__header {
    margin: 0;
    width: 100%;
    height: 200px;
    text-align: center;

    h2 {
        margin: 0 60px;
        margin-bottom: 40px;
        padding-top: 40px;
    }

    @media #{$tablet_breakpoint} {
        h2 {
            font-size: 2em;
        }
    }

    @media #{$desktop_breakpoint} {
        h2 {
            font-size: 2.5em;
            padding-top: 70px;
        }
    }
}


.Home__content {
    width: 100%;
    min-height: calc(100vh - 200px);
    background: $white;

    article {
        padding: 20px;
        text-align: center;
        
        ul {
            margin: 0 auto;
            list-style-type: none;
            font-size: $m_font;
            padding-left: 0;
        }
    }

    @media #{$tablet_breakpoint} {
        article {
            width: 80%;
            margin: 0 auto;

            p {
                font-size: $m_font;
            }
        }
    }

    @media #{$desktop_breakpoint} {
        article {
            max-width: 1200px;
            margin: 0 auto;    
        }
    }
}


.Home__link {
    padding: 8px;
    text-decoration: none;
    background: $primaryColor;
    border-bottom: 4px solid $primaryColor_dark;
    border-radius: 2px 2px 6px 6px;
    color: $white;
    font-weight: bold;
    font-size: $m_font;
    margin: 0 4px;

    &:active {
        border-bottom: none;
        border-radius: 2px;
        position: relative;
        top: 4px;
    }
}