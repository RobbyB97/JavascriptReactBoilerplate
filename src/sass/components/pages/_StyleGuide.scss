#StyleGuide {
    @include page;
}


.StyleGuide__header {
    @include innerContent;
    border-bottom: 1px solid lighten($primaryColor, 15%);
}


.StyleGuide__content {
    @include innerContent;
}