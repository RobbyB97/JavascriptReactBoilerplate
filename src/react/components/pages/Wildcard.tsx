import React from 'react';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';

import { PageType } from './pageType';
import { page_ID__Set } from '../../redux/actions/page';


interface Props extends PageType { }

interface State { }


export class Wildcard extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
    }

    componentWillMount() {
        this.page_ID__Set('Wildcard');
    }

    page_ID__Set = (id: string) => {
        this.props.page_ID__Set(id);
    }


    render() {
        return (
            <div id="Wildcard">

            </div>
        );
    }
}


/* Connect to store */
const mapDispatchToProps = (dispatch: Dispatch<any>) => ({
    page_ID__Set: (id: string) => {
        dispatch(page_ID__Set(id));
    }
});


export default connect(undefined, mapDispatchToProps)(Wildcard);