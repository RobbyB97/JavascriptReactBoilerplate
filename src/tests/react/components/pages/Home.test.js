/* Packages */
import React from 'react';
import {shallow} from 'enzyme';


/* Component */
import {Home} from '../../../../react/components/pages/Home';


/* Test Config */
let component, load, page_ID__Set;
beforeEach(() => {
    load = jest.fn();
    page_ID__Set = jest.fn();
    component = shallow(
        <Home 
            load={load}
            page_ID__Set={page_ID__Set}
        />
    );
});


/* Tests */
test('Render Home page', () => {
    expect(component).toMatchSnapshot();
});


test('Initial state', () => {
    expect(component.state('loaded')).toBe(false);
    expect(component.state('loading')).toBe(false);    
});


test('Load function', () => {
    setTimeout(() => {
        expect(load).toHaveBeenCalled();
    }, 500);
});


test('Page_ID__Set function', () => {
    setTimeout(() => {
        expect(page_ID__Set).toHaveBeenCalled();
    }, 500);
});

