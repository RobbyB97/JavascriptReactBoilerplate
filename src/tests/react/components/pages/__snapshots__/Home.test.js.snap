// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Home page 1`] = `
<div
  data-loaded={false}
  data-loading={false}
  id="Home"
>
  <section
    className="Home__header"
  >
    <h2>
      React Redux Boilerplate v1
    </h2>
    <h4>
      By Robby Bergers
    </h4>
  </section>
  <section
    className="Home__content"
  >
    <article>
      <h3>
        This Boilerplate comes bundled with
      </h3>
      <hr />
      <ul>
        <li>
          React
        </li>
        <li>
          React Router
        </li>
        <li>
          Redux
        </li>
        <li>
          SASS
        </li>
        <li>
          Webpack
        </li>
        <li>
          Node.js
        </li>
        <li>
          Express
        </li>
        <li>
          Source mapping
        </li>
        <li>
          Jest
        </li>
      </ul>
    </article>
    <article>
      <a
        className="Home__link"
        href="https://github.com/RobbyB97/JavascriptReactBoilerplate"
      >
        View on GitHub
      </a>
    </article>
    <article>
      <p>
        Creating software such as this boilerplate takes a lot of time, effort, attention to detail, and the occasional excellent anger management. If you found this boilerplate helpful, please consider 
        <a
          href="https://paypal.me/robbybobby123"
        >
          donating
        </a>
        .
      </p>
      <a
        className="Home__link"
        href="https://paypal.me/robbybobby123"
      >
        Donate
      </a>
    </article>
  </section>
</div>
`;
